app/src/main/java/fho/kdvs/global/MainActivity.kt

```diff
@@ -191,24 +223,44 @@ class MainActivity : DaggerAppCompatActivity() {
         val runnable = object: Runnable {
             override fun run() {
                 handler.postDelayed(this, interval.toLong() * 1000)
-                pb.progress++ // TODO: for some reason this resets to 0 when switching from archive to live
+                pb.progress++
             }
         }
 
         handler.postDelayed(runnable, 0)
     }
 
+    /**
+     * Register a handler that will poll to update UI progress every second.
+     * Register another handler to update persisting exoPlayer position value every second.
+     * */
     private fun initArchiveProgressBar() {
         handler.removeCallbacksAndMessages(null)
 
-        val runnable = object: Runnable {
+        val uiRunnable = object: Runnable {
```

Comment:
These should probably be coroutines instead of `Runnable`s, but that doesn't have to be fixed immediately.

Reply:
I use an anonymous java.lang.Runnable object because the kotlinx coroutines Runnable inline scope cannot self-reference with the "this" keyword, needed to set up the one-second delay.
https://discuss.kotlinlang.org/t/is-this-accessible-in-sams/1477

Also, a Runnable object with the kotlinx.coroutines.Runnable import is just an alias for java.lang.Runnable.

Not sure if this is what you meant by coroutines, though, so feel free to make any improvements.